src/models/query.js

```diff
@@ -0,0 +1,55 @@
+import map from'lodash.map';
+
+
+class Query {
+
+    constructor(name = '') {
+      this._aggregates = this._aggregates || [];
+      this.filters = this.filters || [];
+      this.name = name;
+    }
+    // aggregates
+    addAggregate(aggregate) {
+      // @TODO validate QueryAggregate instance
+      this._aggregates = this._aggregates.concat(aggregate);
+      return this;
+    }
+    getAggregates() {
+      return this._aggregates;
+    }
+
+    // Filters
+    setFilters(filters) {
+      // @TODO validate object type
+      this.filters = filters;
+      return this;
+    }
+    getFilters() {
+      return this.filters;
+    }
+
+    // Sort
+    getName() {
+      return this.name;
+    }
+
+    // Generates the JSON object needed to call the API
+    toJsonAPI() {
+      let aggregates = [];
+      map(this._aggregates, value => {
```

Comment:
you don't need lodash map this.aggregates.map will do the try
Plus, map is used for affectation but you use it like a each
`aggregates = this.aggregates.map(agg => agg.toJsonAPI());`